### PR TITLE
#286 - add search for rooms in "lobby"

### PR DESCRIPTION
**Pull request by dpen2000**
Added search for rooms based on John Resig's LiveSearch with a few modifications.

### CODE REVIEW

File: JabbR/Chat.css [hunk 1]
```diff
@@ -33,6 +33,14 @@
     height: 18px;
 }
 
+#users-filter {
```
Comment (davidfowl) [inline JabbR/Chat.css:36]: Fix formatting.

File: JabbR/Chat.ui.js [hunk 2]
```diff
@@ -24,7 +24,9 @@
         name,
         lastCycledMessage = null,
         $window = $(window),
-        $document = $(document);
+        $document = $(document),
+        roomFilter = null,
```
Comment (davidfowl) [inline JabbR/Chat.ui.js]: All the other jQuery variables start with $

File: JabbR/Chat.ui.js [hunk 3]
```diff
@@ -159,6 +161,10 @@
 
             this.users.removeClass('current')
                       .hide();
+
+            if (this.isLobby()) {
```
Comment (davidfowl) [inline JabbR/Chat.ui.js:164]: Is this supposed to be `if(!this.isLobby())`

File: JabbR/index.htm [hunk 4]
```diff
@@ -198,6 +200,9 @@
         <div id="chat-area">
             <ul id="messages-lobby" class="messages current">
             </ul>
+            <form id="users-filter-form" action="#">
+			<input id="users-filter" class="filter" type="text" />
```
Comment (davidfowl) [inline JabbR/index.htm]: Formatting here as well.

File: JabbR/Chat.ui.js (hunk 3 above)
Comment (dpen2000) [inline JabbR/Chat.ui.js:164]: When we're moving away from lobby, in this function, isLobby returns true and so we hide the textbox for filtering rooms.

File: JabbR/Chat.ui.js [hunk 5]
```diff
@@ -846,6 +856,8 @@
                     $li.addClass('locked');
                 }
             });
+            ui.$roomFilter.update();
+            $roomFilterInput.val('');
```
Comment (davidfowl) [inline JabbR/Chat.ui.js:863]: Why is this in run? What does it do

File: JabbR/Chat.ui.js [hunk 6]
```diff
@@ -741,6 +750,7 @@
                 }
             },
             { unescape: ',/' });
+            ui.$roomFilter = $roomFilterInput.liveUpdate('#users-lobby', true);
```
Comment (dpen2000) [inline JabbR/Chat.ui.js]: This is the line in Run. This is initializing the LiveUpdate plugin which is powering the room search. I think this should probably be moved to the initialize function.

File: JabbR/Chat.ui.js (hunk 5 above)
Comment (dpen2000) [inline JabbR/Chat.ui.js:863]: These two lines are in populateLobbyRooms: I'm updating the LiveUpdate plugin's cache of items (based on the items that have just been populated) and setting the filter back to empty.

Comment (davidfowl) [inline JabbR/Chat.ui.js:863]: Maybe add some comments.

File: JabbR/Chat.ui.js (hunk 6 above)
Comment (davidfowl) [inline JabbR/Chat.ui.js]: me too :). And add some comments.

**Review by Scutua — APPROVED** [review]

**Pull request merged**